File: Lab3/Calculator.go
package main
import (
    "fmt"
    "os"
    "strings"
    "strconv"
    "regexp"
)

func main() {
    input := strings.Split(os.Args[1], " ")
    match, _ := regexp.MatchString(`(?m)\d*(\d\.|\.\d)\d*`, input[0])
    match2, _ := regexp.MatchString(`(?m)\d*(\d\.|\.\d)\d*`, input[2])
    if match == true || match2 == true{
    in1, err1 := strconv.ParseFloat(input[0], 64)
    in2, err2 := strconv.ParseFloat(input[2], 64)
    if err1 == nil && err2 == nil{
        var out float64
        if input[1]== "+" {
            out = in1 + in2
        } else if input[1] == "-" { 
            out = in1 - in2 
        } else if input[1] == "*" { 
            out = in1 * in2 
        } else { 
            out = in1 / in2 
        }

        fmt.Println(out)
    }
    } else{
    in1, err1 := strconv.ParseInt(input[0], 10, 64)
    in2, err2 := strconv.ParseInt(input[2], 10, 64)
    if err1 == nil && err2 == nil{
        var out int64
        if input[1]== "+" {
            out = in1 + in2
        } else if input[1] == "-" { 
            out = in1 - in2 
        } else if input[1] == "*" { 
            out = in1 * in2 
        } else { 
            out = in1 / in2 
        }

        fmt.Println(out)
    }
    }
}

File: Lab3/Calculator.js
var myargs = process.argv[2].split(" ");
const regex = "\\d*(\\.\\d|\\d\\.)\\d*";
if(myargs[0].match(regex) || myargs[2].match(regex)){
	var in1 = parseFloat(myargs[0]);
	var in2 = parseFloat(myargs[2]);
	var out1;
	switch(myargs[1]){
		case "+":
			out1 = in1 + in2; 
			console.log(out1);
			break;
		case "-":
			out1 = in1 - in2;
			console.log(out1);
			break;
		case "*":
			out1 = in1 * in2;
			console.log(out1);
			break;
		case "/":
			out1 = in1 / in2;
			console.log(out1); 
			break;
	}

}
else{
	var in1 = parseInt(myargs[0]);
	var in2 = parseInt(myargs[2]);
	var out;
	switch(myargs[1]){
		case "+":
			out1 = in1 + in2; 
			console.log(out1);
			break;
		case "-":
			out1 = in1 - in2;
			console.log(out1);
			break;
		case "*":
			out1 = in1 * in2;
			console.log(out1);
			break;
		case "/":
			out1 = in1 / in2;
			console.log(out1); 
			break;
	}
	
}